winml/test/model/model_tests.cpp

```diff
@@ -0,0 +1,166 @@
+#include "testPch.h"
+#include "onnxruntime_cxx_api.h"
+#include "test/onnx/TestCase.h"
+#include "test/onnx/heap_buffer.h"
+#include "test/util/include/test/compare_ortvalue.h"
+#include "ort_value_helper.h"
+
+#ifndef BUILD_GOOGLE_TEST
+#error Must use googletest for value-parameterized tests
+#endif
+
+using namespace onnxruntime::test;
+using namespace winml;
+using namespace onnxruntime;
+
+namespace WinML {
+// Global needed to keep the actual ITestCase alive while the tests are going on. Only ITestCase* are used as test parameters.
+std::vector<std::unique_ptr<ITestCase>> ownedTests;
+
+class ModelTest : public testing::TestWithParam<std::tuple<ITestCase*, winml::LearningModelDeviceKind>> {
+ protected:
+  void SetUp() override {
+    std::tie(m_testCase, m_deviceKind) = GetParam();
+    WINML_EXPECT_NO_THROW(m_testCase->GetPerSampleTolerance(&m_perSampleTolerance));
+    WINML_EXPECT_NO_THROW(m_testCase->GetRelativePerSampleTolerance(&m_relativePerSampleTolerance));
+    WINML_EXPECT_NO_THROW(m_testCase->GetPostProcessing(&m_postProcessing));
+  }
+  winml::LearningModelDeviceKind m_deviceKind;
+  ITestCase* m_testCase;
+  std::wstring_convert<std::codecvt_utf8_utf16<wchar_t>> m_converter;
```

Comment:
Better to use `UTF8FromUnicode` in `_winml::Strings` to avoid the STL explosion that happens with codecvt (even though this is test code, better to do it the right way because the code snippet can be copied around and end up in production code)

Reply:
Fixed